src/ethereum_test_specs/base.py

```diff
@@ -46,6 +49,13 @@ def verify_transactions(txs: List[Transaction], result: Result) -> List[int]:
             raise Exception(f"tx expected to fail succeeded: pos={i}, nonce={tx.nonce}")
         elif not tx.error and error:
             raise Exception(f"tx unexpectedly failed: {error}")
+        elif isinstance(tx.error, TransactionException) and error:
```

Comment:
perhaps this whole part should be inside ExceptionMapper. since we compare x and y and have to cover cases
expected y,  got x == y
expected y, got x != y
expected 0, got x 
expected 0, got 0